### PR TITLE
os: fix fdopen invalid file descriptor

### PR DESCRIPTION
**Pull request by deantvv**
Related to #1175

### CODE REVIEW

File: vm/src/stdlib/io.rs [hunk 1]
```diff
@@ -8,6 +8,8 @@ cfg_if::cfg_if! {
         type Offset = i64;
     }
 }
+use crate::stdlib::os::errno_err;
```
Comment (DimitrisJim) [inline vm/src/stdlib/io.rs:11]: Place a `cfg` here and on the next line to get rid of the warnings for windows.

**Comment by DimitrisJim** [comment]
I can't see why the test for wasm is error'ing. Could be a dud.

**Comment by deantvv** [comment]
@youknowone Could you please restart the CI test for macos? It seems that it panic at some weird place which is not related to this PR. Thanks!

**Comment by youknowone** [comment]
@deantvv I think there is an actual bug. Please check @DimitrisJim's review. It also affect WASM.

See Line 58 of the failing test.

```
error[E0432]: unresolved import `crate::stdlib::os`
  --> vm/src/stdlib/io.rs:11:20
   |
11 | use crate::stdlib::os::errno_err;
   |                    ^^ could not find `os` in `stdlib`
```

File: vm/src/stdlib/io.rs (hunk 1 above)
Comment (deantvv) [reply vm/src/stdlib/io.rs]: Thanks for catching this!

**Comment by deantvv** [comment]
> @deantvv I think there is an actual bug. Please check @DimitrisJim's review. It also affect WASM.
> 
> See Line 58 of the failing test.
> 
> ```
> error[E0432]: unresolved import `crate::stdlib::os`
>   --> vm/src/stdlib/io.rs:11:20
>    |
> 11 | use crate::stdlib::os::errno_err;
>    |                    ^^ could not find `os` in `stdlib`
> ```
> 
> For local dev environment,
> 
> ```
> cd wasm/lib && cargo run
> ```

Sorry for this. I fixed this in my local environment but forgot to commit the change.

**Review by youknowone — APPROVED** [review]

**Pull request merged**